draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -2028,1226 +1972,1147 @@ uCDN-A                         dCDN-B                         dCDN-C
+        </t>
+<sourcecode type="json" name="trigger-extension-json"><![CDATA[
+{
+  "generic-trigger-extension-type": "location-policy",
+  "generic-trigger-extension-value": {
+    "locations": [
+      {
+        "action": "allow",
+        "footprints": [{
+          "footprint-type": "countrycode",
+          "footprint-value": [ "us" ]
+        }]
+      },
+      {
+        "action": "deny",
+        "footprints": [{
+          "footprint-type": "countrycode",
+          "footprint-value": [ "ca" ]
+        }]
+      }
+    ]
+  },
+  "mandatory-to-enforce": true,
+  "safe-to-redistribute": true,
+  "incomprehensible": false
+}
+       ]]></sourcecode>
+      </section>
 
-   <t><list style="empty" hangIndent="3">
-   <t><list style="hanging" hangIndent="3"><t hangText="Name: cdn-id">
-   <vspace blankLines="1"/>
-   Description: The CDN PID of the CDN where the error occurred. The "cdn-id" property is used
-                by the originating uCDN or by the propagating dCDN in order to distinguish in which CDN the 
-                error occurred.
-   <vspace blankLines="1"/>
-   Value:  A non-empty JSON string, where the string is a CDN PID as defined in
-                 <xref target="loop-detection-and-prevention"/>
-   <vspace blankLines="1"/>
-   Mandatory: Yes. dCDN may use its own CDN PID if it does not want to expose the CDN PIDs of dCDNs.
-   </t>
+      <section anchor="time-policy" title="TimePolicy Extension">
+        <t>uCDN may wish to perform content management operations on dCDN on a specific schedule.
+           The TimePolicy extension allows uCDN to instruct dCDN to execute the trigger command
+           in a desired time window. For example, a content provider that wishes to pre-populate a new
+           episode at off-peak time so that it would be ready on caches at prime time when the episode
+           is released for viewing. A scheduled operation enables uCDN to direct dCDN in what
+           time frame to execute the trigger.
+        </t>
+        <t>
+           uCDN may wish to schedule a trigger such that dCDN will execute it in local time,
+           as it is measured in each region. For example, uCDN may wish dCDN to pull the content
+           at off-peak hours, between 2AM-4AM, however, as a CDN is distributed across multiple time
+           zones, the UTC definition of 2AM depends on the actual location.
+        </t>
+        <t>We define two alternatives for localized scheduling:<list style="symbols">
+           <t>Regional schedule: When used in conjunction with the Location Policy defined
+              in <xref target="location-policy"/>, uCDN can trigger separate commands for
+              different geographical regions, for each region using a different schedule. This allows
+              uCDN to control the execution time per region.
+           </t>
+           <t>Local Time schedule: We introduce a "local time" version for Internet timestamps
```

Comment:
We also need to remove this paragraph--it pertains to the Local Time Window which I removed earlier (and apparently didn't completely remove).  Maybe we remove the bulleted list, and use the "Regional schedule" approach above as a recommendation for how to remove content in specific geographical regions at specific times.

Reply:
Removed, please review in the upcoming PR. 
As a side note, I am not a fan of the location policy - the notion of location is not properly advertised by dCDN or tied to the configuration metadata in anyway. In the future should be supplanted by footprints and cache buckets.

Reply:
Noted--I see your point!  But we're here now.  I guess the footprint information could at least advise uCDN what footprints are available, even if the Location Policy isn't tied to it.  If uCDN requests a trigger to apply to a region that dCDN doesn't have a presence in, then I suppose the worst that could happen is that the trigger is a no-op.